### DISCUSSION TITLE
Document activity tracking / audit

### DISCUSSION THREAD

**Discussion by slysiou** [appwrite/appwrite#4901] [Ideas]
Hi,
I am creating a function to add document activity tracking. But, I think it would be great to add this feature as an option on each collection.

The idea is to add $updatedBy on each document.
And create a collection_history recording each activities (create, delete, update) with the state of the document information.

Anyone else see the same need ?

**Comment by netapy** [comment]
> $updatedBy on each document

Definitely, most service use cases using Appwrite need this at some point.

> a collection_history recording each activities (create, delete, update) with the state of the document information

I'm less sure this would be necessary in most cases, it would probably just bloat the storage space in my opinion.

=> Looking forward to an $updatedBy attribute in the future...

**Comment by slysiou** [comment]
collection_history should be an option and not by default.
take a look at: 
https://hibernate.org/orm/envers/
https://github.com/Blackburn-Labs/parse-auditor
"This can be very helpful for apps that need to adhere to regulations like healthcare apps which need to adhere to HIPAA standards. This can also be used for apps that have a high level of data sensitivity like legal or FinTech apps."

**Comment by stnguyen90** [comment]
> I am creating a function to add document activity tracking. 

FYI, Appwrite does have some audit logs. In the Console, you can see this in the "Activity" tabs for a Document. It looks like they're only accessible with an Admin account (i.e. not via server SDK). More details on the API can be found here: 

https://github.com/appwrite/appwrite/blob/81b08de407a4dbcde70ad591ca9015f18e1e8a4c/app/controllers/api/databases.php#L2122

**Comment by stnguyen90** [comment]
> The idea is to add $updatedBy on each document.

If you're interested in this, please make sure to 👍🏼 [this](https://github.com/appwrite/appwrite/issues/3623) issue.

**Comment by Ricolo53** [comment]
+1 about state of the document on audit worker. We need to know what the change has been, and not only action name
